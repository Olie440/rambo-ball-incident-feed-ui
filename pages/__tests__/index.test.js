import React from 'react';
import { shallow } from 'enzyme';

import { IndexPage, mapStateToProps } from '../index.js';
import MockRedux from '~/__mocks__/mock-redux';
import getGameData from '~/redux/actions/get-game-data';

jest.mock('~/redux/actions/get-game-data');

describe('Index Page', () => {
    let mockRedux;

    beforeEach(() => {
        mockRedux = MockRedux();
    });

    it('renders correctly', () => {
        const element = shallow(<IndexPage game={mockRedux.state.game}/>);
        expect(element).toMatchSnapshot();
    });

    it('maps the redux state to props correctly', () => {
        const result = mapStateToProps(mockRedux.state);
        expect(result).toEqual({
            game: mockRedux.state.game
        });
    });

    describe('getInitalProps', () => {
        let mockContext, getGameDataAction;

        beforeEach(() => {
            getGameDataAction = jest.fn();

            getGameData.mockReset();
            getGameData.mockReturnValue(getGameDataAction);

            mockContext = {
                reduxStore: mockRedux,
                query: {
                    id: '1234'
                }
            };
        });

        it('returns an empty object', async () => {
            const actual = await IndexPage.getInitialProps(mockContext);
            expect(actual).toEqual({});
        });

        it('creates a getGameData action using the injected redux', async () => {
            await IndexPage.getInitialProps(mockContext);
            const { dispatch, getState } = mockRedux;

            expect(getGameData).toHaveBeenCalledWith(dispatch, getState);
        });

        it('it calls the created getGameData action', async () => {
            await IndexPage.getInitialProps(mockContext);
            expect(getGameDataAction).toHaveBeenCalledWith('1234');
        });

        it('does not create an action when query.id is missing', async () => {
            mockContext.query = {};
            await IndexPage.getInitialProps(mockContext);

            expect(getGameData).not.toHaveBeenCalled();
            expect(getGameDataAction).not.toHaveBeenCalled();
        });

        it('returns an empty object when query.id is missing', async () => {
            mockContext.query = {};
            const actual = await IndexPage.getInitialProps(mockContext);
            
            expect(actual).toEqual({});
        });
    });
});