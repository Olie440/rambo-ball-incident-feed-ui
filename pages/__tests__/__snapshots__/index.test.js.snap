// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders correctly 1`] = `
<section
  className="indexPage"
>
  <IncidentList
    incidents={
      Array [
        Object {
          "id": "203d7268-1d13-49ac-a879-52bd6d071d8a",
          "timestamp": 1547825693,
          "type": "GOAL_MISSED_AWAY",
        },
        Object {
          "id": "a87d40a1-8027-444c-82ba-7f07cdb85081",
          "timestamp": 1547825691,
          "type": "GOAL_ATTEMPT_AWAY",
        },
        Object {
          "id": "683fd087-8f90-41e8-a85d-e53a4be9c128",
          "timestamp": 1547825069,
          "type": "AWAY_POSSESSION",
        },
        Object {
          "id": "3896af84-1886-48e3-b396-c5b1592151b1",
          "timestamp": 1547825062,
          "type": "GOAL_SCORED_HOME",
        },
        Object {
          "id": "41e418c9-d9e0-4ff8-b857-fb6db5d5adc5",
          "timestamp": 1547824982,
          "type": "HOME_POSSESSION",
        },
        Object {
          "id": "b72a9ebd-721c-49fa-aaef-45ea710ec87d",
          "timestamp": 1547824682,
          "type": "KICK_OFF",
        },
      ]
    }
    pageSize={4}
  />
</section>
`;
